gold-plugins/two-step-auth.js

```diff
@@ -111,41 +131,90 @@ exports.commands = {
+				return this.sendReply("Check your email for verification - it will have you enter a command to verify that you own this email.");
+			} else if (targets[0] === 'authenticator') {
+				let twoAuthData = twoFactor.generateSecret({name: 'Gold PS', account: user.userid});
+				user.tempTwoAuth = twoAuthData.secret;
+				let uri = "otpauth://totp/Gold-PS:" + user.userid + "?secret=" + twoAuthData.secret + "&issuer=Gold-PS";
+				let qrImg = "https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=" + uri;
+				let reply = "|modal||html|Please enter the following code into your authenticator application or scan the QR code.<br />";
+				reply += "Key: " + twoAuthData.secret + "<br />";
+				reply += "<img src=\"" + qrImg + "\" height=\"166\" width=\"166\"><br />";
+				reply += "Once you have added the key to your authenticator, please verify it by running ";
+				reply += "the following command:<br />";
+				reply += "<code>/twostep verify [code from your authenticator]</code>";
+				return user.popup(reply);
+			}
 		},
 		login: function (target, room, user, connection) { // undocumented
 			if (!user.pendingRename) return this.errorReply("This is a secret command.");
 			if (!target) return false;
 			if (target === 'restart') return Gold.TwoStepAuth.sendCodePrompt(user);
-			if (isNaN(target)) return false;
-			target = parseInt(target);
+			if (isNaN(target) && target !== 'R' && target !== '<-') return false;
 			if (!user.codeAttempt) user.codeAttempt = [];
+			if (target === '<-') {
+				user.codeAttempt.splice(-1);
+				return Gold.TwoStepAuth.sendCodePrompt(user);
+			}
 			user.codeAttempt.push(target);
-			if (user.codeAttempt.length >= 5) {
-				Gold.TwoStepAuth.verifyCode(toId(user.pendingRename.targetName), user, Number(user.codeAttempt.join('')), connection);
+			if (user.codeAttempt.length >= 6) {
+				Gold.TwoStepAuth.verifyCode(toId(user.pendingRename.targetName), user, user.codeAttempt.join(''), connection);
 				user.codeAttempt = [];
 			} else {
 				Gold.TwoStepAuth.sendCodePrompt(user);
 			}
 		},
 		verify: function (target, room, user) { // undocumented
 			if (!user.named) return this.errorReply("You must be logged in to use this command.");
-			if (!target) return false;
-			let verified = (Number(target) === user.twostepEmail.code);
-			if (verified) {
-				Gold.userData[user.userid].email = user.twostepEmail.email;
-				Gold.saveData();
-				return this.sendReply("Two-step authentication has been officially setup for your account.");
-			} else {
-				return this.errorReply("Unfortunately, you entered the wrong verification code.");
+			if (user.twostepEmail) {
+				if (!target) return false;
+				let verified = (Number(target) === user.twostepEmail.code);
+				if (verified) {
+					Gold.userData[user.userid].email = user.twostepEmail.email;
+					Gold.saveData();
+					return this.sendReply("Two-step authentication has been officially setup for your account.");
```

Comment:
... as opposed to unofficially?

This was shitty wording to begin with; remove "officially"